ral-facilities/datagateway
# Pressing 'Enter' to search and removing unnecessary text #689

## Description
This PR makes a couple of changes to improve dg search.
- Searching can now be initiated by typing in a search query and pressing Enter. Previously, searching could only be initiated solely by pressing the Search button. This can be done in the 'Search text' field and both of the date input fields
- Before a search was started, the text "Fill out form to the left and then click search." was displayed at the bottom of the page. This is being removed as it is deemed self-explanatory.

The majority of the work in this PR involves moving functions and tests and remaking them to account for these moves. For example, the logic to initiate a search has been moved to a parent component so that it can be called by the search input box, search button, start date and end date boxes. So the majority of the code that has been "removed" has merely been "moved".

## Testing instructions
Navigate to DataGateway search and try searching using the Enter key. Type in a search term and then press Enter. It should have the same functionality as pressing the Search button next to the search input field. Also try searching using date ranges (both valid and invalid values) and using Enter to start search. Experiment with different types of queries with different tables and date ranges. Also check that "Fill out form to the left and then click search." is no longer displayed on the page.

- [x] Test searches with Enter key
- [x] Ensure "Fill out form to the left and then click search." is not displayed

## Agile board tracking
Closes #689
